src/main/java/algorithms/curated170/medium/shortestdistancetotargetcolor/ShortestDistanceToTargetColorPreprocessing.java

```diff
@@ -0,0 +1,72 @@
+package algorithms.curated170.medium.shortestdistancetotargetcolor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ShortestDistanceToTargetColorPreprocessing {
+
+    private static final int NOT_FOUND = Integer.MAX_VALUE;
+    private int[][] closestDistTable;
+
+    public List<Integer> shortestDistanceColor(int[] colors, int[][] queries) {
+
+        assignClosestLeftColors(colors);
+
+        compareClosestRightColors(colors);
+
+        return handleQueriesLookup(queries);
+    }
+
+    private void assignClosestLeftColors(int[] colors) {
+        closestDistTable = new int[colors.length][];
```

Comment:
What does the second dimension of closestDistTable store?

Reply:
Closest index of each color.